pilot/pkg/serviceregistry/external/conversion.go

```diff
@@ -51,7 +51,21 @@ func convertServices(serviceEntry *networking.ServiceEntry) []*model.Service {
 	for _, host := range serviceEntry.Hosts {
 		if len(serviceEntry.Addresses) > 0 {
 			for _, address := range serviceEntry.Addresses {
-				if _, _, cidrErr := net.ParseCIDR(address); cidrErr == nil || net.ParseIP(address) != nil {
+				if ip, network, cidrErr := net.ParseCIDR(address); cidrErr == nil {
```

Comment:
should we complain about CIDR during validation until we support it in our config generation?